src/error.rs

```diff
@@ -8,33 +8,30 @@ use std::{error, fmt, io, result, sync};
 /// `mongodb::Error`.
 pub type Result<T> = result::Result<T, Error>;
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum MaliciousServerErrorType {
     InvalidRnonce,
     InvalidServerSignature,
     NoServerSignature,
 }
 
-impl fmt::Display for MaliciousServerErrorType {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+impl MaliciousServerErrorType {
+    fn to_str(&self) -> &'static str {
+        use self::MaliciousServerErrorType::*;
         match *self {
-            MaliciousServerErrorType::InvalidRnonce => {
-                fmt.write_str(
-                    "The server returned an invalid rnonce during authentication",
-                )
-            }
-            MaliciousServerErrorType::InvalidServerSignature => {
-                fmt.write_str(
-                    "The server returned an invalid signature during authentication",
-                )
-            }
-            MaliciousServerErrorType::NoServerSignature => {
-                fmt.write_str("The server did not sign its reponse during authentication")
-            }
+            InvalidRnonce => "The server returned an invalid rnonce during authentication",
+            InvalidServerSignature => "The server returned an invalid signature during authentication",
+            NoServerSignature => "The server did not sign its reponse during authentication",
```

Comment:
Totally unrelated to the changes you made, but I just realized that "response" is misspelled.

Reply:
Indeed, let me fix that too!